File: newcliwebjsuper-master/src/app/modules/producto/models/filter-producto.ts
class CodigoBarras {
    codigo: string = ""
    constructor(codigo: string) {
        this.codigo = codigo;
    }
}
class ProductoPadre {
    nombre: string = "";
}
class Producto {
    nombre: string = "";
    codigoEspecial: string = "";
    id: number = 0;
    codigoBarras: Array<CodigoBarras> = new Array<CodigoBarras>(new CodigoBarras(""));
    productoPadre: ProductoPadre = new ProductoPadre();
}
export class FilterProducto {
    producto: Producto = new Producto();
    filterStock: number = 0;
    showStock: boolean = false;
    extraFilter: boolean = false;
    familias: [] = [];
    sucursales: [] =[];
    activo: number = 1;
    tipo: number = 1 // por defecto productos simples
}

File: BackJSuper-master/persistence/src/main/java/ar/com/jsuper/domain/utils/FilterProducto.java
package ar.com.jsuper.domain.utils;

import java.util.Set;

import ar.com.jsuper.domain.Familias;
import ar.com.jsuper.domain.Producto;
import ar.com.jsuper.domain.Sucursales;

public class FilterProducto {

    private Producto producto;
    private Boolean showStock;
    private Integer activo; //0= inactivos, 1: activos, 2 : todos
    private Integer filterStock; //0= todos, 1: stockMinimo, 2 : puntoReposicion
    private Integer tipo; //1= simples, 2: compuestos, 0 : todos
    private Set<Familias> familias;
    private Set<Sucursales> sucursales;

    public Producto getProducto() {
        return producto;
    }

    public void setProducto(Producto producto) {
        this.producto = producto;
    }

    public Set<Familias> getFamilias() {
        return familias;
    }

    public void setFamilias(Set<Familias> familias) {
        this.familias = familias;
    }

    public Integer getActivo() {
        return activo;
    }

    public void setActivo(Integer activo) {
        this.activo = activo;
    }

    public Integer getTipo() {
        return tipo;
    }

    public void setTipo(Integer tipo) {
        this.tipo = tipo;
    }

    public Set<Sucursales> getSucursales() {
        return sucursales;
    }

    public void setSucursales(Set<Sucursales> sucursales) {
        this.sucursales = sucursales;
    }

    public Boolean getShowStock() {
        return showStock;
    }

    public void setShowStock(Boolean showStock) {
        this.showStock = showStock;
    }

    public Integer getFilterStock() {
        return filterStock;
    }

    public void setFilterStock(Integer filterStock) {
        this.filterStock = filterStock;
    }

}
